### PR TITLE
Make internal links go to correct page positions

### LINKED ISSUE #237: Link to new view does not go back to top of the page
## Problem

If we click on a link in the middle of the page to go to another view, the scroll is not reset and we end up in the middle of the destination page.

## Suggestion

- use the `fragment` option of [reitit.frontend.easy/href](https://cljdoc.org/d/metosin/reitit/0.7.0-alpha3/api/reitit.frontend.easy#href) of version `0.7.0-alpha3`

### PR DESCRIPTION
**Pull request by flybot-nam-nguyenhoai**
## Closes #237

- [x] Links to posts on the website now navigate to the top of the corresponding posts, using URL `#` fragments.
    - [x] Blog post links navigate to the top of their respective blog post pages.
    - [x] Links to posts on other pages navigate to the top of the corresponding post containers.

### CODE REVIEW

**Comment by flybot-nam-nguyenhoai** [comment]
Here's an interesting bug: When a post link on the profile page is clicked, the new view does not automatically scroll to the correct post; the browser has to be refreshed for the link to go to the correct position.

**Comment by skydread1** [comment]
You can use a `reg-fx` to scroll to an element after the view has been rendered:

```clojure
(rf/reg-fx
 :fx.app/scroll-to
 (fn [_ _]
   (reagent/after-render #(when-let [el (.getElementById js/document "app")]
                      (.scrollIntoView el)))))
```

Then add this fx in the event:

```clojure
(rf/reg-event-fx
 :evt.page/set-current-view
 (fn [{:keys [db]} [_ new-match]]
   {:db (assoc db :app/current-view new-match)
    :fx [[:fx.app/scroll-to]]}))
```

This always go to top of the page but params can be added to provide the `dom-id` to scroll to

**Comment by flybot-nam-nguyenhoai** [comment]
The above `.scrollIntoView` snippet is obtained from https://clojurians.slack.com/archives/C0620C0C8/p1619879858211700.

(Exact search phrase: `Say I want to scroll to an element after a page is rendered.`)

File: client/web/src/flybot/client/web/core/db/fx.cljs [hunk 1]
```diff
@@ -8,6 +8,17 @@
...
             [reagent.core :as reagent]))
 
+;;; -------- Routing ---------
+
+;; URL document fragments
+
+(rf/reg-fx
+ :fx.app/scroll-to
+ (fn [fragment]
+   (reagent/after-render #(let [el (or (.getElementById js/document fragment)
```
Comment (skydread1) [inline client/web/src/flybot/client/web/core/db/fx.cljs:18]: the `or` is not correct, "app" case will not work.

File: client/web/src/flybot/client/web/core/db/event.cljs [hunk 2]
```diff
@@ -142,10 +142,12 @@
...
 ;; ---------- View ----------
 
-(rf/reg-event-db
+(rf/reg-event-fx
  :evt.page/set-current-view
- (fn [db [_ new-match]]
-   (assoc db :app/current-view new-match)))
+ (fn [{:keys [db]} [_ new-match]]
+   {:db (assoc db :app/current-view new-match)
+    :fx (when-let [fragment (:fragment new-match)]
```
Comment (skydread1) [inline client/web/src/flybot/client/web/core/db/event.cljs:149]: `:fx` value can't be nil:

```
re-frame: ":fx" effect expects a seq, but was given  null
```

**Review by skydread1 — CHANGES_REQUESTED** [review]

File: client/web/src/flybot/client/web/core/db/fx.cljs (hunk 1 above)
Comment (flybot-nam-nguyenhoai) [reply client/web/src/flybot/client/web/core/db/fx.cljs:18]: Fixed.

File: client/web/src/flybot/client/web/core/db/event.cljs (hunk 2 above)
Comment (flybot-nam-nguyenhoai) [reply client/web/src/flybot/client/web/core/db/event.cljs]: Fixed.

**Review by skydread1 — APPROVED** [review]
Good

**Pull request merged**